setup.py

```diff
@@ -502,7 +502,7 @@ def get_sphinx_theme_version() -> str:
     'paramiko',
     'pipdeptree',
     'pre-commit',
-    'pylint>=2.7.0',
+    'pylint~=2.7.4',
```

Comment:
Is this a required change or drive-by refactoring?

Reply:
If we don't do it, we will need to fix new issues reported by Pylint 2.8.0 released yesterday. 


General context: 

All the reqular PR's are using the current 'constraints'. But whenever we modify setup.py, such PR will automatically attempt to run "eager upgrade" to get latest versions of everything (including pylint new version in this case). This is our way to - pretty much automatically - upgrade all such constraints and detect any future incompatibilies early (but without affecting the regular PRs).

Context for pylint: 

We've already had similar problem caused by new pylint versions (new errors reported on already 'good' code whenever setup.py changedf) so I'd rather fix it now to minor version and upgrade it separately in a separate PR (especially that they are preparing to release a major 3.0 upgrade which might require a bit more 'overhaul'.